client/modules/wj-css/src/components/_sheaf.scss

```diff
@@ -0,0 +1,219 @@
+    margin: 0 0.25rem;
+    color: var(--colcode-content);
+  }
+
+  .cm-button {
+    border: 1px solid var(--colcode-border);
+    background: none;
+  }
+
+  .cm-textfield {
+    border: 1px solid var(--colcode-border);
+  }
+
+  &.cm-focused ::selection,
+  ::selection {
+    background-color: var(--colcode-selection);
+    color: var(--colcode-content);
+  }
+
+  .cm-line {
+    ::selection,
+    &::selection {
+      color: inherit !important;
+    }
+  }
+
+  .cm-activeLine {
+    background-color: var(--colcode-hover);
+  }
+
+  .cm-selectionBackground,
+  &.cm-focused .cm-selectionBackground {
+    background-color: var(--colcode-selection);
+  }
+
+  .cm-selectionMatch {
+    background-color: var(--colcode-selection);
+  }
+
+  .cm-searchMatch {
+    background-color: var(--colcode-selection);
+    border-radius: 0.125rem;
+
+    &.selected {
+      background-color: var(--colcode-selection);
+      box-shadow: 0 0 0 0.075rem var(--colcode-accent);
+    }
+  }
+
+  .cm-matchingBracket,
+  .cm-nonmatchingBracket {
+    background-color: var(--colcode-hover);
+    outline: 1px solid var(--colcode-selection);
+  }
+
+  &.cm-focused {
+    outline: none;
+
+    .cm-cursor {
+      border-left-color: var(--colcode-accent);
+    }
+
+    @include tolerates-motion {
+      // Makes the mouse cursor smooth
+      .cm-cursorLayer {
+        animation: cubic-bezier(0.95, 0, 0.05, 1) cm-blink 1.2s infinite;
+      }
+
+      .cm-cursor {
+        transition: left 0.05s ease-out, top 0.05s ease-out;
+      }
+    }
+  }
+}
+
+.cm-editor .cm-tooltip {
+  border: 1px solid var(--colcode-border);
+  background-color: var(--colcode-background);
+  border-radius: 0.25rem;
+  @include shadow(4);
+
+  @include tolerates-motion {
+    animation: cm-tooltip-fadein 0.125s 1 0s backwards ease-out;
+  }
+
+  .cm-completionInfo {
+    max-width: 30rem;
+  }
+
+  .cm-completionIcon {
+    line-height: 0;
+    font-size: 1.25em;
+    width: 1em;
+    padding-right: 0.25em;
+    color: var(--colcode-keyword);
+    opacity: 1;
+  }
+
+  &.cm-tooltip-autocomplete {
+    animation: none;
+
+    > ul {
+      font-family: var(--sheaf-font);
+      font-size: 0.875em;
+      color: var(--colcode-content);
+
+      > li {
+        line-height: 1.5;
+        &[aria-selected] {
+          background-color: var(--colcode-selection);
+          color: var(--colcode-content);
+        }
+      }
+    }
+  }
+}
+
+.cm-editor {
+  .cm-panels-top {
+    border-bottom: 2px solid var(--colcode-border);
+  }
+
+  .cm-panels-bottom {
+    border-top: 2px solid var(--colcode-border);
+  }
+
+  .cm-panels {
+    background-color: var(--col-background);
+    color: var(--colcode-content);
+  }
+}
+
+.cm-editor .cm-panel {
+  &.cm-panel-lint ul {
+    max-height: 16rem;
+    outline: none;
+    padding-right: 0.5rem;
+
+    // use duplicate selector so we get more specificity
+    // the CodeMirror base theme overrides otherwise
+    &:focus [aria-selected][aria-selected] {
```

Comment:
Wow I've literally never thought to do this

Reply:
What does the duplicate selector do?

Reply:
CSS selectors have a certain way that they cascade and conflict (override) each other. This is their _specificity_, and it's a source of eternal headache. If two selectors have the same specificity, they'll go off of declaration order. Otherwise, the higher specificity wins. Usually, the more complex a selector is, the higher specificity it has. So by repeatedly selecting `[aria-selected]`, we increase the specificity and override the CodeMirror base theme.

Reply:
Interesting